docs/build-speed-android.md

```diff
@@ -0,0 +1,132 @@
+# ./gradlew <task> -PreactNativeArchitectures=x86_64
+reactNativeArchitectures=armeabi-v7a,arm64-v8a,x86,x86_64
+```
+
+Once you build a **release version** of your app, don't forget to remove those flags as you want to build an apk/app bundle that works for all the ABIs and not only for the one you're using in your daily development workflow.
+
+## Use a compiler cache
+
+If you're running frequent native builds, you might benefit from using a compiler cache.
+Specifically you can use two type of caches: local compiler caches and distributed compiler caches.
+
+### Local caches
+
+We suggest to use [**ccache**](https://ccache.dev/) to cache the compilation of your native builds.
+Ccache works by wrapping the C++ compilers, storing the compilation results, and skipping the compilation
+if an intermediate compilation result was originally stored.
+
+To install it, you can follow the [official installation instructions](https://github.com/ccache/ccache/blob/master/doc/INSTALL.md).
+
+On Mac OS, we recommend to install ccache with `brew install ccache`.
+Once installed you can configure it as follows:
+
+```
+ln -s ccache /usr/local/bin/gcc
+ln -s ccache /usr/local/bin/g++
+ln -s ccache /usr/local/bin/cc
+ln -s ccache /usr/local/bin/c++
+```
+
+This will create symbolic links to `ccache` inside the `/usr/local/bin/` which are called `gcc`, `g++`, `cc` and `c++`.
+
+This works as long as `/usr/local/bin/` comes first than `/usr/bin/` inside your `$PATH` variable, which is the default.
+
+You can verify that it works using the `which` command:
+
+```
+$ which gcc
+/usr/local/bin/gcc
+```
+
+If the results is `/usr/local/bin/gcc`, then you're effectively calling `ccache` which will wrap the `gcc` calls.
+
+You can then do two Android clean builds. You will notice that the second build was way faster than the first one (it should take seconds rather than minutes).
+While building, you can verify that `ccache` works correctly and check the cache hits/miss rate `ccache -s`
+
+```
+$ ccache -s
+Summary:
+  Hits:             196 /  3068 (6.39 %)
+    Direct:           0 /  3068 (0.00 %)
+    Preprocessed:   196 /  3068 (6.39 %)
+  Misses:          2872
+    Direct:        3068
+    Preprocessed:  2872
+  Uncacheable:        1
+Primary storage:
+  Hits:             196 /  6136 (3.19 %)
+  Misses:          5940
+  Cache size (GB): 0.60 / 20.00 (3.00 %)
+```
+
+Should you need to wipe your cache, you can do so with `ccache --clear`
+
+#### Using this approach on a CI
+
+Ccache uses the `/Users/$USER/Library/Caches/ccache` folder on macOS to store the cache.
+Therefore you could save & restore this folder also on CI to speedup your builds.
+
+However, there are a couple of things to be aware:
+
+1. On CI, we recommend to do a full clean build, to avoid poisoned cache problems. If you follow the approach mentioned in the previous paragraph, you should be able to parallelize the native build on 4 different ABIs and you will most likely not need `ccache` on CI.
```

Comment:
Suggest speedup might still be worth it, there is a ccache-action that works great, and you can specify a versioned cache key to wipe out the cache if you suspect a poisoned cache. All of this in action:

https://github.com/invertase/react-native-firebase/blob/ef7895cd49eed3a4d103df624e52315250c68ab3/.github/workflows/tests_e2e_ios.yml#L70-L74